### ISSUE TITLE
Talon on Mac causing some issue with Safari causing it to reload tabs on launch

### ISSUE THREAD

**Issue by BrokenFlows** [talonvoice/talon#335]
![Screenshot 2021-06-30 at 12 10 56](https://user-images.githubusercontent.com/21959941/123951215-78381900-d99c-11eb-82e6-31d5d588e48d.png)

I'm not sure what the cause is, but I have been able to repeatedly cause issues in Safari, resulting in reloading tabs, when launching Talon.

I can currently reproduce this behaviour with Talon version 0.1.5-468 on Safari version 14.1 (16611.1.21.161.3).
I realise that I haven't given a whole lot of information, I wasn't sure what else I could put i, but if you have any questions I'm happy to try answer them.

**Comment by lunixbochs** [comment]
Can you show me the output of `ui.apps()`? Or maybe `[(app.name, app.bundle, app.background) for app in ui.apps()]` I used to see this issue myself, but I added a fix in the beta for it (sometime around 0.1.5-300), and I don't see it anymore. It's related to querying the background (non-ui) safari processes for accessibility information, apparently that crashes them.

**Comment by BrokenFlows** [comment]
Sure, below is the output of `[(app.name, app.bundle, app.background) for app in ui.apps()]` from the REPL. There's a bunch of WebKit stuff below, which might give plenty of opportunities for such an issue to arise.
```
[('Control Strip', 'com.apple.controlstrip', True), 
('loginwindow', 'com.apple.loginwindow', True), 
('AXVisualSupportAgent', 'com.apple.accessibility.AXVisualSupportAgent', True), 
('Mail', 'com.apple.mail', False), 
('Audacity', 'org.audacityteam.audacity', False), 
('Reminders', 'com.apple.reminders', False), 
('Calendar', 'com.apple.iCal', False), 
('SystemUIServer', 'com.apple.systemuiserver', True), 
('Dock', 'com.apple.dock', True), 
('Control Centre', 'com.apple.controlcenter', True), 
('ViewBridgeAuxiliary', 'com.apple.ViewBridgeAuxiliary', True), 
('QuickLookUIService (PID 1120)', 'com.apple.quicklook.QuickLookUIService', True), 
('Notification Centre', 'com.apple.notificationcenterui', False), 
('UIKitSystem', 'com.apple.UIKitSystemApp', True), 
('imklaunchagent', '', True), 
('com.apple.PressAndHold', 'com.apple.PressAndHold', True), 
('EmojiFunctionRowIM', 'com.apple.inputmethod.EmojiFunctionRowItem', True), 
('Wi-Fi', 'com.apple.wifi.WiFiAgent', True), 
('Dock Extra', 'com.apple.dock.extra', True), 
('Dock Extra', 'com.apple.dock.external.extra.arm64', True), 
('Mail Networking', 'com.apple.WebKit.Networking', True), 
('Location Menu', 'com.apple.locationmenu', True), 
('Finder', 'com.apple.finder', False), 
('BitBar', 'com.matryer.BitBar', True), 
('TextInputMenuAgent', 'com.apple.TextInputMenuAgent', True), 
('AirPlayUIAgent', 'com.apple.AirPlayUIAgent', True), 
('CheatSheet', 'com.mediaatelier.CheatSheet', True), 
('Audio Hijack 3 Schedule Helper', 'com.rogueamoeba.audiohijack3.schedulehelper', True), 
('SteerMouse Manager', 'jp.plentycom.boa.SteerMouse', True), 
('Script Menu', 'com.apple.ScriptMenuApp', True), 
('Alfred', 'com.runningwithcrayons.Alfred', False), 
('Thyme', 'com.joaomoreno.Thyme', True), 
('Deliveries Express', 'com.junecloud.mac.Deliveries.Updater', True), 
('FuzzyTime', 'com.mrcs.FuzzyTime', True), 
('Wallet', 'com.apple.passd', True), 
('CrossOver CD Helper', 'com.codeweavers.CrossOverCDHelper', True), 
('FastScripts', 'com.red-sweater.fastscripts', True), 
('Rocket', 'net.matthewpalmer.Rocket', True), 
('photolibraryd', 'com.apple.photolibraryd', True), 
('studentd', 'com.apple.studentd', True), 
('UserEventAgent', '', True), 
('Bartender 4', 'com.surteesstudios.Bartender', False), 
('CalendarAgent', 'com.apple.CalendarAgent', True), 
('ArqMonitor', 'com.haystacksoftware.ArqMonitor', True), 
('Finder Networking', 'com.apple.WebKit.Networking', True), 
('com.apple.speech.speechsynthesisd', '', True), 
('universalAccessAuthWarn', 'com.apple.accessibility.universalAccessAuthWarn', True), 
('Webcam Settings', 'com.mactaris.Webcam-Settings', True), 
('SpeechSynthesisServerXPC', 'com.apple.speech.SpeechSynthesisServerXPC', True), 
('1Password Extension Helper', '2BUA8C4S2C.com.agilebits.onepassword7-helper', True), 
('UserNotificationCenter', 'com.apple.UserNotificationCenter', True), 
('1Password (Safari)', 'com.agilebits.onepassword7.1PasswordSafariAppExtension', True), 
('Safari Networking', 'com.apple.WebKit.Networking', True), 
('SiriNCService', 'com.apple.SiriNCService', True), 
('Vimari (Safari)', 'net.televator.Vimari.SafariExtension', True), 
('Safari', 'com.apple.Safari', False), 
('CacheDeleteExtension Networking', 'com.apple.WebKit.Networking', True), 
('com.apple.Safari.SandboxBroker (Safari)', 'com.apple.Safari.SandboxBroker', True), 
('AccessibilityVisualsAgent', 'com.apple.AccessibilityVisualsAgent', True), 
('DockHelper', 'com.apple.dock.helper', True), 
('StoreUIService', 'com.apple.AppStoreDaemon.StoreUIService', True), 
('Siri', 'com.apple.Siri', True), 
('CoreLocationAgent', 'com.apple.CoreLocationAgent', True), 
('SoftwareUpdateNotificationManager', 'com.apple.SoftwareUpdateNotificationManager', True), 
('talagent', 'com.apple.talagent', True), 
('TextInputSwitcher', 'com.apple.TextInputSwitcher', True), 
('OSDUIHelper', 'com.apple.OSDUIHelper', True), 
('coreautha', 'com.apple.LocalAuthentication.UIAgent', True), 
('LookupViewService (Safari)', 'com.apple.LookupViewService', True), 
('Front and Center', 'co.hypercritical.Front-and-Center', True), 
('Notes', 'com.apple.Notes', False), 
('com.apple.appkit.xpc.openAndSavePanelService (Safari)', 'com.apple.appkit.xpc.openAndSavePanelService', True), 
('QuickLookUIService (com.apple.appkit.xpc.openAndSavePanelService (Safari))', 'com.apple.quicklook.QuickLookUIService', True), 
('Dropbox', 'com.getdropbox.dropbox', True), 
('PowerChime', 'com.apple.PowerChime', True), 
('AppSSOAgent', 'com.apple.AppSSOAgent', True), 
('Dropbox Activity Provider', 'com.dropbox.activityprovider', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('QuickLookUIService (com.apple.appkit.xpc.openAndSavePanelService (ideaMaker))', 'com.apple.quicklook.QuickLookUIService', True), 
('com.apple.appkit.xpc.openAndSavePanelService (ideaMaker)', 'com.apple.appkit.xpc.openAndSavePanelService', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('PhotosReliveWidget', 'com.apple.Photos.PhotosReliveWidget', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('Preview', 'com.apple.Preview', False), 
('FirefoxCP RDD Process', 'org.mozilla.plugincontainer', True), 
('FirefoxCP WebExtensions', 'org.mozilla.plugincontainer', True), 
('SceneKitQLPreviewExtension (Finder)', 'com.apple.SceneKitQLPreviewExtension', True), 
('FirefoxCP Privileged Content', 'org.mozilla.plugincontainer', True), 
('FirefoxCP Web Content', 'org.mozilla.plugincontainer', True), 
('Terminal', 'com.apple.Terminal', False), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('Parcel Mac Widget', 'com.mr-brightside.ParcelOSX.Widget', True), 
('FirefoxCP Web Content', 'org.mozilla.plugincontainer', True), 
('Firefox', 'org.mozilla.firefox', False), 
('FirefoxCP Web Content', 'org.mozilla.plugincontainer', True), 
('FirefoxCP Web Content', 'org.mozilla.plugincontainer', True), 
('FirefoxCP Web Content', 'org.mozilla.plugincontainer', True), 
('FirefoxCP Web Content', 'org.mozilla.plugincontainer', True), 
('com.apple.appkit.xpc.openAndSavePanelService (Fusion 360)', 'com.apple.appkit.xpc.openAndSavePanelService', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('FirefoxCP Web Content', 'org.mozilla.plugincontainer', True), 
('QuickLookUIService (com.apple.appkit.xpc.openAndSavePanelService (Fusion 360))', 'com.apple.quicklook.QuickLookUIService', True), 
('Talon', 'com.talonvoice.Talon', True), 
('com.apple.WebKit.Networking', 'com.apple.WebKit.Networking', True), 
('Speech Synthesis Server', 'com.apple.speech.synthesis.SpeechSynthesisServer', True), 
('Keychain Circle Notification', 'com.apple.security.Keychain-Circle-Notification', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('NowPlayingTouchUI', 'com.apple.NowPlayingTouchUI', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('CoreServicesUIAgent', 'com.apple.coreservices.uiagent', True), 
('nbagent', 'com.apple.nbagent', True), 
('Fusion 360', 'com.autodesk.fusion360', False), 
('Spotlight', 'com.apple.Spotlight', True), 
('storeuid', 'com.apple.storeuid', True), 
('MobileDeviceUpdater', 'com.apple.MobileDeviceUpdater', True), 
('Ultimaker Cura', 'nl.ultimaker.cura', False), 
('ideaMaker', 'com.raise3d.ideaMaker', False), 
('Home', 'com.apple.Home', False), 
('Safari Web Content', 'com.apple.WebKit.WebContent', True), 
('Safari Web Content', 'com.apple.WebKit.WebContent', True), 
('Safari Web Content', 'com.apple.WebKit.WebContent', True), 
('Safari Web Content', 'com.apple.WebKit.WebContent', True), 
('Safari Web Content', 'com.apple.WebKit.WebContent', True), 
('Safari Web Content', 'com.apple.WebKit.WebContent', True), 
('Safari Web Content', 'com.apple.WebKit.WebContent', True), 
('Safari Web Content (Prewarmed)', 'com.apple.WebKit.WebContent', True), 
('Safari Web Content', 'com.apple.WebKit.WebContent', True), 
('Safari Web Content', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('Steam Helper', 'com.valvesoftware.steam.helper', True), 
('Steam', 'com.valvesoftware.steam', False), 
('Steam Helper', 'com.valvesoftware.steam.helper', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('AirDrop', 'com.apple.share.AirDrop.send', True), 
('QuickLookUIService', 'com.apple.quicklook.QuickLookUIService', True), 
('1Password 7', 'com.agilebits.onepassword7', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('Dropbox Finder Extension', 'com.getdropbox.dropbox.garcon', True), 
('Code', 'com.microsoft.VSCode', False), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True), 
('com.apple.WebKit.WebContent', 'com.apple.WebKit.WebContent', True)]
```

**Comment by lunixbochs** [comment]
fixed in 0.1.5-472
